utils/stall_free.hh

```diff
@@ -179,19 +180,18 @@ future<> clear_gently(T& o) noexcept {
 
 template <SharedPointer T>
 future<> clear_gently(foreign_ptr<T>& o) noexcept {
-    return smp::submit_to(o.get_owner_shard(), [&o] {
-        if (o.unwrap_on_owner_shard().use_count() == 1) {
-            return internal::clear_gently(const_cast<std::remove_const_t<typename foreign_ptr<T>::element_type>&>(*o));
-        }
-        return make_ready_future<>();
-    });
+    co_await smp::submit_to(o.get_owner_shard(), seastar::coroutine::lambda([&o] -> future<> {
+        auto wrapped = o.release();
+        co_await internal::clear_gently(wrapped);
+    }));
```

Comment:
Is the internal coroutine necessary? And the external one for that matter.

Reply:
Te alternative is do_with to the wrapped alive while being cleared gently.

Reply:
But o.unwrap_on_current_shared() is a smart pointer, that you can move to the continuation.

Reply:
right. I've gotten rusty with continuations

Reply:
Done in [71b19ac](https://github.com/scylladb/scylladb/commit/71b19acc30f32721a11d4e2adc24ef4b6c478cda)